File: Hash Table/2423. Remove Letter To Equalize Frequency/equalFrequency.go
func equalFrequency(word string) bool {
	digit := [26]int{}
	for i := 0; i < len(word); i++ {
		digit[word[i]-'a']++
	}

	for i := 0; i < len(word); i++ {
		index := word[i] - 'a'
		digit[index]--
		freqs := 0
		temp := true
		for j := 0; j < 26; j++ {
			if digit[j] != 0 {
				if freqs == 0 {
					freqs = digit[j]
				} else if freqs != digit[j] {
					temp = false
				}
			}
		}
		if temp {
			return true
		}
		digit[index]++
	}

	return false
}

File: Hash Table/2423. Remove Letter To Equalize Frequency/equalFrequency.java
class Solution {
    public boolean equalFrequency(String word) {
        int[] digit=new int[26];
        for(int i=0;i<word.length();i++){
            digit[word.charAt(i) - 'a']++;
        } 
        for(int i=0;i<word.length();i++){
            int index=word.charAt(i) -'a';
            digit[index]--;
            int freqs=0;
            boolean temp=true;
            for(int j=0;j<26;j++){
                if(digit[j]!=0){
                    if(freqs==0){
                        freqs=digit[j];
                    }
                    if(freqs!=digit[j]){
                        temp=false;
                    }
                }
            }
            if(temp==true){
                return true;
            }
            digit[index]++;
        }
        return false;
    }
}